Suppress "Not all requests have been executed" error when running individual test test_create_scan

### Describe the bug

Running `test_create_scan` individually causes an `AssertionError` from `responses` due to unmatched mocked requests, even though the test passes functionally. This is misleading and not needed for this specific test.

**Error log:**

```
  File "/home/zoheb/workspace/blueapi/tests/conftest.py", line 479, in mock_numtracker_server
    with responses.RequestsMock() as requests_mock:
         ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/zoheb/workspace/blueapi/.venv/lib/python3.12/site-packages/responses/__init__.py", line 996, in __exit__
    self.stop(allow_assert=success)
  File "/home/zoheb/workspace/blueapi/.venv/lib/python3.12/site-packages/responses/__init__.py", line 1233, in stop
    raise AssertionError(
AssertionError: Not all requests have been executed [('POST', 'https://numtracker-example.com/graphql'), ('POST', 'https://numtracker-example.com/graphql'), ('POST', 'https://numtracker-example.com/graphql'), ('POST', 'https://numtracker-example.com/graphql')]
```

---

### Steps To Reproduce

1. Run:

   ```bash
   pytest tests/unit_tests/client/test_numtracker.py::test_create_scan  
   ```

2. Observe the `AssertionError` from unexecuted mocked requests.

---

### Acceptance Criteria
* [ ] Use `assert_all_requests_are_fired=False`.

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -444,7 +444,7 @@ def mock_numtracker_server() -> Iterable[responses.RequestsMock]:
     }
     empty_response = {}
 
-    with responses.RequestsMock() as requests_mock:
+    with responses.RequestsMock(assert_all_requests_are_fired=False) as requests_mock:
         requests_mock.add(
             responses.POST,
             url="https://numtracker-example.com/graphql",

